core/src/main/java/com/linecorp/armeria/server/http/dynamic/DynamicHttpService.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.http.dynamic;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.linecorp.armeria.common.http.DefaultHttpResponse;
+import com.linecorp.armeria.common.http.HttpMethod;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.http.AbstractHttpService;
+import com.linecorp.armeria.server.http.HttpService;
+
+/**
+ * An {@link HttpService} that serves dynamic contents.
+ */
+public final class DynamicHttpService extends AbstractHttpService {
+    /**
+     * A {@link List} of {@link DynamicHttpFunctionEntry}, sorted in ascending order of it's path skeleton.
+     * */
+    private List<DynamicHttpFunctionEntry> entries;
+
+    DynamicHttpService(List<DynamicHttpFunctionEntry> entries) {
+        this.entries = entries;
+    }
+
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        HttpMethod method = req.method();
+        String path = ctx.mappedPath();
+
+        Optional<MappedDynamicFunction> mapped = entries.stream().map(e -> e.bind(method, path))
+                                                        .filter(Objects::nonNull)
+                                                        .findFirst();
```

Comment:
Given that this may be a hot spot, could we avoid using a stream? How about making `entries` member field a `DynamicHttpFunctionEntry[]` so that it's as cheap as possible to loop over it?